test/StackTrace/dotChainNameHint.js

```diff
@@ -0,0 +1,41 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+if (typeof (WScript) != "undefined") {
+    WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+}
+
+var tests = [
+    {
+        name: "OS#14711878: Throw SOE in Parser::ConstructNameHint",
+        body: function ()
+        {
+            assert.throws(function () {
+                const M = 1e4;
+                var u;
+                for (var i = 0; i < M; i++) {
+                    u = u + ".prototype";
+                }
+                eval(u);
+            }, Error, "Should SOE in Parser::ConstructNameHint", "Out of stack space");
+        }
+    },
+    {
+        name: "OS#14711878: SOE in Emit",
+        body: function ()
+        {
+            assert.throws(function () {
+                const M = 1e3;
+                var u;

```

Comment:
Should these `u` variables be initialized to something, or is the eval string supposed to be `.prototype.prototype.[...]` rather than `foo.prototype.prototype.[...]`?

Reply:
This was based on the OSS-Fuzz test case. It ends up expanding to `undefined.prototype.prototype...`